lib/Report/SummaryReport.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+namespace Xibo\Report;
+
+use Xibo\Factory\LayoutFactory;
+use Xibo\Factory\MediaFactory;
+use Xibo\Factory\SavedReportFactory;
+use Xibo\Exception\InvalidArgumentException;
+use Xibo\Factory\DisplayFactory;
+use Xibo\Service\ConfigServiceInterface;
+use Xibo\Service\DateServiceInterface;
+use Xibo\Service\LogServiceInterface;
+use Xibo\Service\SanitizerServiceInterface;
+use Xibo\Storage\StorageServiceInterface;
+use Xibo\Storage\TimeSeriesStoreInterface;
+
+/**
+ * Class SummaryReport
+ * @package Xibo\Report
+ */
+class SummaryReport implements ReportInterface
+{
+    use ReportTrait;
+
+    /**
+     * Report Constructor.
+     * @param \Xibo\Helper\ApplicationState $state
+     * @param StorageServiceInterface $store
+     * @param TimeSeriesStoreInterface $timeSeriesStore
+     * @param LogServiceInterface $log
+     * @param ConfigServiceInterface $config
+     * @param DateServiceInterface $date
+     * @param SanitizerServiceInterface $sanitizer
+     * @param DisplayFactory displayFactory
+     * @param MediaFactory mediaFactory
+     * @param LayoutFactory layoutFactory
+     * @param SavedReportFactory savedReportFactory
+     */
+    public function __construct($state, $store, $timeSeriesStore, $log, $config, $date, $sanitizer, $displayFactory, $mediaFactory, $layoutFactory, $savedReportFactory)
+    {
+        $this->setCommonDependencies($state, $store, $timeSeriesStore, $log, $config, $date, $sanitizer, $displayFactory, $mediaFactory, $layoutFactory, $savedReportFactory);
+    }
+
+    /** @inheritdoc */
+    public function getReportForm()
+    {
+        return 'summary-report-form';
+    }
+
+    /** @inheritdoc */
+    public function generateSavedReportName($filterCriteria)
+    {
+
+        if ($filterCriteria['type'] == 'layout') {
+            $layout = $this->layoutFactory->getById($filterCriteria['layoutId']);
+            $saveAs = ucfirst($filterCriteria['filter']). ' report for Layout '. $layout->layout;
+
+        } else {
+            $media = $this->mediaFactory->getById($filterCriteria['mediaId']);
+            $saveAs = ucfirst($filterCriteria['filter']). ' report for Media '. $media->name;
+        }
+
+        return $saveAs;
+    }
+
+    /** @inheritdoc */
+    public function getSavedReportResults($savedreportId)
+    {
+
+        $savedReport = $this->savedReportFactory->getById($savedreportId);
+
+        // Open a zipfile and read the json
+        $zipFile = $this->getConfig()->getSetting('LIBRARY_LOCATION') . $savedReport->storedAs;
```

Comment:
this bit of logic here probably belongs in the report service rather than the specific summary report - meaning we should enforce all reports save/retrieve their data in the same way